### PR TITLE
Drawtext with Polyzones for atms

### PR DESCRIPTION
**Pull request by Tomboru**
**Describe Pull request**
With this pull request you can just go to an atm
and press (e) - it uses polyzones and the build-in drawtext of QB core.

This is useful because you don't have to use /atm anymore.
The resmon is always on 0.00ms while in polyzone maybe 0.01ms

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? yes!
- Does your code fit the style guidelines? yes
- Does your PR fit the contribution guidelines? yes

### CODE REVIEW

**Review by skyrocker203 — COMMENTED** [review]
tested it out, but when i come near the ATMS, nothing happens. got the atm card in inventory

**Comment by tafabdel** [comment]
working fine for me

File: fxmanifest.lua [hunk 1]
```diff
@@ -11,7 +11,14 @@ server_scripts {
     'server/main.lua'
 }
 
-client_script 'client/main.lua'
+client_script {
+	'@PolyZone/client.lua',
+    '@PolyZone/BoxZone.lua',
+    '@PolyZone/EntityZone.lua',
```
Comment (Z3rio) [inline fxmanifest.lua:17]:
```suggestion
```
BoxZones nor EntityZones ever get used in this script.

File: fxmanifest.lua [hunk 2]
```diff
@@ -11,7 +11,14 @@ server_scripts {
     'server/main.lua'
 }
 
-client_script 'client/main.lua'
+client_script {
+	'@PolyZone/client.lua',
```
Comment (Z3rio) [inline fxmanifest.lua:15]:
```suggestion
    '@PolyZone/client.lua',
```
Incorrect linting / spacing

File: fxmanifest.lua [hunk 3]
```diff
@@ -11,7 +11,14 @@ server_scripts {
...
+    '@PolyZone/BoxZone.lua',
+    '@PolyZone/EntityZone.lua',
+    '@PolyZone/CircleZone.lua',
+    '@PolyZone/ComboZone.lua',
+	'client/main.lua'
```
Comment (Z3rio) [inline fxmanifest.lua:20]:
```suggestion
    'client/main.lua'
```
Incorrect linting / spacing

File: config.lua [hunk 4]
```diff
@@ -8,3 +8,126 @@ Config.ATMModels = {
...
+		vector3(2559.0522, 389.47443, 107.62291),
+		vector3(1703.3152, 4934.0527, 41.063587),
+		vector3(1735.0105, 6410.01, 34.03717),
+		vector3(2683.592, 3286.3,  54.241077),
+		vector3(1968.3923, 3743.0784, 31.343689),
+		vector3(540.22064, 2671.683, 41.15644),
+		vector3(-42.9, -1108.35, 26.44),
+		vector3(315.08, -593.68, 43.28),
+		vector3(-468.68, 45.81, 52.41)
+}
```
Comment (Z3rio) [inline config.lua:133]:
```suggestion
Config.ATMcoords = {
	vector3(-301.65726, -829.5886, 31.419765),
	vector3(-303.2257, -829.3121, 31.419765),
	vector3(-204.0193, -861.0091, 29.271332),
	vector3(118.64156, -883.56946, 30.13945),
	vector3(24.5933, -945.543, 28.333046),
	vector3(5.686035, -919.9551, 28.48088),
	vector3(296.17563, -896.2318, 28.290146),
	vector3(296.8775, -894.3196, 28.261478),
	vector3(147.47305, -1036.2175, 28.367783),
	vector3(145.83922, -1035.6254, 28.367783),
	vector3(111.38856, -774.84015, 30.437658),
	vector3(114.54742, -775.9721, 30.417364),
	vector3(-256.6386, -715.88983, 32.7883),
	vector3(-259.27673, -723.2652, 32.701546),
	vector3(-254.52185, -692.8869, 32.578255),
	vector3(-27.890343, -724.10895, 43.22287),
	vector3(228.03244, 337.85013, 104.50133),
	vector3(158.79654, 234.74516, 105.643265),
	vector3(527.77765, -160.66086, 56.136715),
	vector3(-57.170288, -92.37918, 56.750687),
	vector3(89.813385, 2.880325, 67.35214),
	vector3(285.3485, 142.97507, 103.16232),
	vector3(357.12845, 174.08362, 102.059654),
	vector3(1137.8113, -468.86255, 65.698654),
	vector3(1167.06, -455.6541, 65.818565),
	vector3(1077.7786, -776.96643, 57.256516),
	vector3(289.52997, -1256.7876, 28.440575),
	vector3(289.26785, -1282.3204, 28.65519),
	vector3(-165.58443, 234.76587, 93.92897),
	vector3(-1044.466, -2739.6414, 8.12406),
	vector3(-1205.3783, -326.5286, 36.85104),
	vector3(-1206.1417, -325.03165, 36.85104),
	vector3(-846.6537, -341.50903, 37.668503),
	vector3(-847.204, -340.42908, 37.6793),
	vector3(-720.6288, -415.52432, 33.97996),
	vector3(-867.013, -187.99278, 36.882175),
	vector3(-867.97455, -186.34193, 36.882175),
	vector3(-1415.4801, -212.33244, 45.49542),
	vector3(-1430.6633, -211.35867, 45.47162),
	vector3(-1410.7357, -98.927895, 51.39701),
	vector3(-1410.183, -100.64539, 51.396523),
	vector3(-1282.0983, -210.55992, 41.43031),
	vector3(-1286.7037, -213.78275, 41.43031),
	vector3(-1289.742, -227.16498, 41.43031),
	vector3(-1285.1365, -223.94215, 41.43031),
	vector3(-712.93567, -818.4827, 22.740658),
	vector3(-710.08276, -818.4756, 22.736336),
	vector3(-617.80347, -708.8591, 29.043213),
	vector3(-617.80347, -706.8521, 29.043213),
	vector3(-614.5187, -705.5981, 30.223999),
	vector3(-611.8581, -705.5981, 30.223999),
	vector3(-660.67633, -854.48816, 23.456635),
	vector3(-537.8052, -854.93567, 28.275429),
	vector3(-594.61444, -1160.8519, 21.333511),
	vector3(-596.12506, -1160.8503, 21.3336),
	vector3(-526.7791, -1223.3737, 17.45272),
	vector3(-1569.8396, -547.0309, 33.932163),
	vector3(-1570.7653, -547.7035, 33.932163),
	vector3(-1305.7078, -706.6881, 24.314468),
	vector3(-1315.416, -834.431, 15.952328),
	vector3(-1314.466, -835.6913, 15.952328),
	vector3(-2071.9285, -317.2862, 12.318085),
	vector3(-821.89355, -1081.5546, 10.136639),
	vector3(-1110.2284, -1691.1538, 3.378483),
	vector3(-2956.8481, 487.21576, 14.478001),
	vector3(-2958.977, 487.30713, 14.478001),
	vector3(-2974.5864, 380.12692, 14.0),
	vector3(-1091.8875, 2709.0535, 17.919415),
	vector3(-2295.8525, 357.93475, 173.60143),
	vector3(-2295.0693, 356.2556, 173.60143),
	vector3(-2294.2998, 354.6056, 173.60143),
	vector3(-3144.8875, 1127.811, 19.838036),
	vector3(-3043.8347, 594.16394, 6.732796),
	vector3(-3241.4546, 997.9085, 11.548369),
	vector3(2563.9995, 2584.553, 37.06807),
	vector3(2558.3242, 350.988, 107.597466),
	vector3(156.18863, 6643.2, 30.59372),
	vector3(173.8246, 6638.2173, 30.59372),
	vector3(-282.7141, 6226.43, 30.496475),
	vector3(-95.870285, 6457.462, 30.473938),
	vector3(-97.63721, 6455.732, 30.467934),
	vector3(-132.66629, 6366.8765, 30.47258),
	vector3(-386.4596, 6046.4106, 30.473991),
	vector3(1687.3951, 4815.9, 41.006466),
	vector3(1700.6941, 6426.762, 31.632965),
	vector3(1822.9714, 3682.5771, 33.267452),
	vector3(1171.523, 2703.1394, 37.147697),
	vector3(1172.4573, 2703.1394, 37.147697),
	vector3(238.26779, 217.10918, 105.40615),
	vector3(238.69781, 216.18698, 105.40615),
	vector3(237.83775, 218.03137, 105.40615),
	vector3(237.40773, 218.95358, 105.40615),
	vector3(236.9777, 219.87578, 105.40615),
	vector3(264.86896, 209.94864, 105.40615),
	vector3(265.21695, 210.9048, 105.40615),
	vector3(265.913, 212.81714, 105.40615),
	vector3(266.26102, 213.77332, 105.40615),
	vector3(380.65576, 322.8424, 102.56634),
	vector3(1153.1111, -326.90186, 68.20503),
	vector3(33.19432, -1348.8058, 28.49696),
	vector3(130.57912, -1292.3688, 28.271421),
	vector3(130.15036,  -1291.6261, 28.271421),
	vector3(129.69753, -1290.8418, 28.271421),
	vector3(-57.402237, -1751.7471, 28.420937),
	vector3(-718.26135, -915.71277, 18.21553),
	vector3(-273.36655, -2024.2079, 29.169643),
	vector3(-262.36078, 2012.054, 29.169643),
	vector3(-1391.3445, -589.86273, 29.315836),
	vector3(-1827.6887, 784.465, 137.31522),
	vector3(-3040.2046, 593.29694, 6.908859),
	vector3(-3240.028, 1008.5453, 11.830639),
	vector3(2559.0522, 389.47443, 107.62291),
	vector3(1703.3152, 4934.0527, 41.063587),
	vector3(1735.0105, 6410.01, 34.03717),
	vector3(2683.592, 3286.3,  54.241077),
	vector3(1968.3923, 3743.0784, 31.343689),
	vector3(540.22064, 2671.683, 41.15644),
	vector3(-42.9, -1108.35, 26.44),
	vector3(315.08, -593.68, 43.28),
	vector3(-468.68, 45.81, 52.41)
}
```
Incorrect linting / spacing

File: client/main.lua [hunk 5]
```diff
@@ -87,6 +123,7 @@ end)
 -- Callbacks
 
 RegisterNUICallback("NUIFocusOff", function()
+	InATM = false
```
Comment (Z3rio) [inline client/main.lua:126]:
```suggestion
    InATM = false
```
Incorrect linting / spacing

File: client/main.lua [hunk 6]
```diff
@@ -37,22 +54,41 @@ RegisterNetEvent('qb-atms:client:updateBankInformation', function(banking)
...
+						event = 'qb-atms:server:enteratm',
+						type = 'server',
+						icon = "fas fa-credit-card",
+						label = "Use ATM",
+					},
+				},
+				distance = 1.5,
+		})
+	end
+end)
```
Comment (Z3rio) [inline client/main.lua:91]:
```suggestion
local NewZones = {}
CreateThread(function()
    if not Config.UseTarget then
        for k, v in pairs(Config.ATMcoords) do
            NewZones[#NewZones+1] = CircleZone:Create(vector3(v.x, v.y, v.z), 1.5, {
                useZ = true,
                debugPoly = false,
                name = k,
            })
        end

        local combo = ComboZone:Create(NewZones, {name = "RandomZOneName", debugPoly = false})
        combo:onPlayerInOut(function(isPointInside, point, zone)
            if isPointInside then
                exports["qb-core"]:DrawText("[E] Use ATM")
                Listen4Control()
            else
                exports["qb-core"]:HideText()
                listen = false
            end
        end)
    else
        exports['qb-target']:AddTargetModel(Config.ATMModels, {
            options = {
                {
                    event = 'qb-atms:server:enteratm',
                    type = 'server',
                    icon = "fas fa-credit-card",
                    label = "Use ATM",
                },
            },
            distance = 1.5,
        })
    end
end)
```
Incorrect linting / spacing

File: client/main.lua [hunk 7]
```diff
@@ -21,9 +21,26 @@ local function PlayATMAnimation(animation)
...
+            if IsControlJustPressed(0, 38) then -- E
+                    exports["qb-core"]:KeyPressed()
+                    TriggerServerEvent("qb-atms:server:enteratm")
+                listen = false
+                break
+            end
+            Wait(1)
+        end
+    end)
+end
```
Comment (Z3rio) [inline client/main.lua:38]:
```suggestion
local listen = false
local function Listen4Control()
    listen = true

    CreateThread(function()
        while listen do
            if IsControlJustPressed(0, 38) then -- E
                exports["qb-core"]:KeyPressed()
                TriggerServerEvent("qb-atms:server:enteratm")
                listen = false
                break
            end
            Wait(1)
        end
    end)
end
```
Fix linting / spacing

**Review by Z3rio — CHANGES_REQUESTED** [review]

**Comment by Tomboru** [comment]
Thanks to @tafabdel for going through the code and make it more readable.
This ATM script should still work completely fine.

**Review by Tomboru — COMMENTED** [review]
Thanks to @tafabdel for going through the code and make it more readable.
This ATM script should still work completely fine.

**Review by Tomboru — COMMENTED** [review]
All changes are approved from my side, and already applied!

**Pull request closed without merging**